### PR TITLE
AGS: Make NumberPtr a class instead of a union

### PR DESCRIPTION
**Pull request by lephilousophe**
It's illegal to reference an inactive member of a union and the types are not layout compatible (and even not of the same size). Instead, store every value in a integer of pointer size and cast pointers to and from it.
This should really fix bug [#13557](https://bugs.scummvm.org/ticket/13557).

### CODE REVIEW

**Review by bluegr — APPROVED** [review]
LGTM

**Review by dwatteau — APPROVED** [review]
Thanks a lot! It reminds me of the https://github.com/cmus/cmus/pull/941 fix I needed for cmus on ppc64 some years ago.

I confirm that it does fix bug [#13557](https://bugs.scummvm.org/ticket/13557): Maniac Mansion Deluxe now starts on PS3!

So it looks good to me, both for the main and the 2.9.0 branches.

(By the way, I do have a quad G5 at my parents', which could be useful to do some ppc64 tests and debugging as well… In fact I had in mind that this PS3 bug could come from the 64-bit big-endian thing, but I had no idea where to look at. Having a "real" ppc64 system on hand, with an easy access to a debugger and so on, could have helped identifying the problem a bit more easily… ScummVM on ppc64 is in my todo-list for 2025…)

**Comment by lephilousophe** [comment]
Thanks for the test.
I will wait for a review by AGS engine maintainers before merging.

**Comment by PaulGilbert** [comment]
Gave it a quick visual lookover, and the changes look good. If you could also just change the comment above the class so it doesn't say "union" anymore as well

**Comment by lephilousophe** [comment]
Indeed, I didn't read the comment. Thanks!

**Comment by tag2015** [comment]
Looks good, I've done some tests and found no issues.
Thanks!

File: engines/ags/engine/script/cc_instance.cpp
Comment (dwatteau) [inline engines/ags/engine/script/cc_instance.cpp:1225]:
```suggestion
						error("Unhandled 64-bit pointer result from plugin method call");
```

We might as well fix that typo along the way :p

**Comment by lephilousophe** [comment]
Great! Merging.

**Pull request merged**
